traffic_ops/traffic_ops_golang/routes.go

```diff
@@ -122,6 +122,7 @@ func Routes(d ServerData) ([]Route, []RawRoute, http.Handler, error) {
 		{1.1, http.MethodPut, `divisions/{id}$`, api.UpdateHandler(division.GetRefType(), d.DB), auth.PrivLevelOperations, Authenticated, nil},
 		{1.1, http.MethodPost, `divisions/?$`, api.CreateHandler(division.GetRefType(), d.DB), auth.PrivLevelOperations, Authenticated, nil},
 		{1.1, http.MethodDelete, `divisions/{id}$`, api.DeleteHandler(division.GetRefType(), d.DB), auth.PrivLevelOperations, Authenticated, nil},
+		{1.1, http.MethodGet, `divisions/name/{name}/?(\.json)?$`, division.GetName(d.DB.DB), auth.PrivLevelReadOnly, Authenticated, nil},
```

Comment:
Note: this pr could only be this line:
`{1.1, http.MethodGet, `divisions/name/{name}?$`, api.ReadHandler(division.GetRefType(), d.DB), auth.PrivLevelReadOnly, Authenticated, nil}`

I also think it shouldn't have the .json ending as I don't think the perl endpoint supports it with a path parameter.

Reply:
yeah confirmed we don't need this to be a valid url:
'https://localhost:8443/api/1.3/divisions/name/Northeast/.json'

Reply:
`https://localhost:8443/api/1.3/divisions/name/Northeast.json` works in Perl though

Reply:
I tried changing this to `api.ReadHandler(division.GetRefType()`, but even with `/?(\.json)?$`, going to `/api/1.3/divisions/name/Northeast.json` (which works in Perl) returns `[]`.

I had to change `TODivision.Read` to add this to the top:
```
	if strings.HasSuffix(parameters["name"], ".json") {
		parameters["name"] = parameters["name"][:len(parameters["name"]) - len(".json")]
	}
```

I tried changing `shared_handlers` `GetStringKey` to do that, but it didn't work, and I'm not sure why, and I don't have time to investigate, adding it to `Read` worked.